plugins/org.jkiss.dbeaver.ext.gaussdb/src/org/jkiss/dbeaver/ext/gaussdb/model/GaussDBDialect.java

```diff
@@ -73,40 +106,77 @@ public void generateStoredProcedureCall(StringBuilder sql, DBSProcedure proc,
         boolean first = true;
         for (DBSProcedureParameter parameter : inParameters) {
             String typeName = parameter.getParameterType().getFullTypeName();
-            switch (parameter.getParameterKind())
-            {
-            case INOUT:
-            case IN:
-                if (!first) {
-                    sql.append(", ");
-                }
-                if (castParams) {
-                    sql.append("cast(").append(namedParameterPrefix).append(CommonUtils.escapeIdentifier(parameter.getName()))
-                        .append(" as ").append(typeName).append(")");
-                } else {
-                    sql.append(namedParameterPrefix).append(CommonUtils.escapeIdentifier(parameter.getName()));
-                }
-                break;
-            case RETURN:
-                continue;
-            default:
-                if (isStoredProcedureCallIncludesOutParameters()) {
+            switch (parameter.getParameterKind()) {
+                case INOUT:
+                case IN:
                     if (!first) {
                         sql.append(", ");
                     }
                     if (castParams) {
-                        sql.append("cast(?").append(" as ").append(typeName).append(")");
+                        sql.append("cast(").append(namedParameterPrefix).append(CommonUtils.escapeIdentifier(parameter.getName()))
+                            .append(" as ").append(typeName).append(")");
                     } else {
-                        sql.append("?");
+                        sql.append(namedParameterPrefix).append(CommonUtils.escapeIdentifier(parameter.getName()));
+                    }
+                    break;
+                case RETURN:
+                    continue;
+                default:
+                    if (isStoredProcedureCallIncludesOutParameters()) {
+                        if (!first) {
+                            sql.append(", ");
+                        }
+                        if (castParams) {
+                            sql.append("cast(?").append(" as ").append(typeName).append(")");
+                        } else {
+                            sql.append("?");
+                        }
                     }
-                }
-                break;
+                    break;
             }
             first = false;
         }
     }
+
+    @Override
+    public String getQuotedIdentifier(String str, boolean forceCaseSensitive, boolean forceQuotes) {
+        if (isQuotedIdentifier(str)) {
+            // Already quoted
+            return str;
+        }
+        String[][] quoteStrings;
+        String databaseCompatibleMode = "";
+        GaussDBDataSource dataSource = getDataSource();
+        if (dataSource instanceof DatabaseCompatibilityProvider) {
+            DatabaseCompatibilityProvider compatibilityProvider = (DatabaseCompatibilityProvider) dataSource;
```

Comment:
Maybe using GaussDBDataSource.getDatabaseCompatibleMode if OK. Do not need to check `dataSource instanceof DatabaseCompatibilityProvider`